src/common/time/CheckTimer.h

```diff
@@ -0,0 +1,37 @@
+// Copyright (C) 2024 European Spallation Source, ERIC. See LICENSE file
+//===----------------------------------------------------------------------===//
+///
+/// \file
+///
+/// \brief platform independent equivalent replacement of wrapper for the cheap and fast time stamp counter (TSC)
+///
+/// Uses the standard chrono library to provide a high resolution timer.
+//===----------------------------------------------------------------------===//
+
+#pragma once
+
+#include <chrono>
+#include <cstdint>
+
+class CheckTimer {
```

Comment:
As I can see we have the TSCTimer for similar purpose. What is the reason of the implementation of this new CheckTimer? It does something differently or better? Shall I streamline the other detectors and outdate the TSC?

Reply:
I added this for portability/accuracy. 

The `TSCTimer` uses a x86 specific instruction to read a hardware clock and compares it to a fixed number of cycles to decide when to trigger; the instruction used does not exist on other hardware [compare x86_64 and arm8](https://godbolt.org/#z:OYLghAFBqd5QCxAYwPYBMCmBRdBLAF1QCcAaPECAMzwBtMA7AQwFtMQByARg9KtQYEAysib0QXACx8BBAKoBnTAAUAHpwAMvAFYTStJg1DIApACYAQuYukl9ZATwDKjdAGFUtAK4sGe1wAyeAyYAHI%2BAEaYxCAA7KQADqgKhE4MHt6%2BekkpjgJBIeEsUTHxdpgOaUIETMQEGT5%2BXLaY9nkM1bUEBWGR0XG2NXUNWc0KQ93BvcX9sQCUtqhexMjsHAD0mwDUxJhM6FuObFvjrAlbaF6C0VsAtDtXCltMBFsJxKirCkrEWwASAC8TBoAIKnRzILYAfShwVoUxhW0eeGAIQOtAEwC2GKMO3QBAUyAgADdUHh0HMtiZYlZQVskQwUqjMAcEHhSNjUCYAMy0kH0mFMBQsREw0kGRz0GEQcxmYj4wmyrYgKlmMw8gAiTFlEAxC1V6u5GvQOrZczmPL59N2BGWDC2UGRzPRmM5RjmeqpsTcDsdjJRaLdWJxwDmZqp3LcPJ93LMFt5wJB1I1icTwVeLCYwQgFppifp9KYXiIWwYqAA7hGNXiCUT41aCza7aWK5bU7ENRwFrROABWXh%2Bbi8VCcKOWawnJYrTCq7k8UgETRdhYAaxA3O5ADoNzvd3uAGz6TiSAdL0gjji8BQgDQLpcLOCwJBoFgJOjRciUF9v%2BgxZAGIwoQIYgrhXPg6AIaJrwgCIzwiYJagAT04ed4OYYhEIAeQibQKkXIdSBfNhBEwhhaGQjgtFILAIi8YA3DEWhrwIrBMyMcRKN4fBdkqYlMGYqjMFUCpizWed01aM94QiYgkI8LAz2AvAWBQ3g%2BOICJkkwDVMDY4B4SMe8%2BAMYAFAANTwTBy0whJGFUmRBBEMR2CkBz5CUNQz10ZoAOMaxrH0PAImvSAFlQBJ2mY4d1OIcl%2BPgBZykqZwIFcEYmlIQIpiKEpsmSVIBHSvLcjSHocv6MZWjwqoJiKyq2hqroyr6GIxlqzxGj0U46mamZWsSqdVgkbs%2B1PTjz04LZVAADn3W590kC5fIdYDQMpCBcEIEhZy4OZeHwrRzVINcN23Pdzo3Q8ew4E9SEHKiLyvG8704h8YEQFBUFfd8yAoCBvx%2BkBahYYlptuf9DGAW5VoYMCaFoSDiGg2DxrQpD7LRjDsNwhx7KIxgCFI8izxouiGNoJj7NYyGOKo7jqr4gTeCEkTIPsiTrqo6TZIw%2BS1iopSVII9TNKUHS9IM0BXuMphTIsqybLsgj%2BEc0RxFclX3JUdRxt0Mx9EhlB/MsQLgoS88IrSKLzxiuKQpzFoGpStKOqyABOW8ssKFqQGkHICvSV2/A9xJ8vaXrcukJL2k6YYg5AEPo8anrsp9qP2syYPb26yZvb632BuWIbdqPDh%2Bzus8L2eYgQbBpglshlaQNh9bNqIX5zDnPaXsO1d1y3C6LtL277uHTgntvA7l1LsxeBUrgNFvUeJsvHvp/UlJnEkIA%3D%3D%3D) and the fixed number of clock ticks is not scaled to the running CPU frequency, nor take into account frequency changes.

`CheckTimer` has the same interface as `TSCTimer` but doesn't require an [x86 system to compile](https://godbolt.org/#z:OYLghAFBqd5QCxAYwPYBMCmBRdBLAF1QCcAaPECAMzwBtMA7AQwFtMQByARg9KtQYEAysib0QXACx8BBAKoBnTAAUAHpwAMvAFYTStJg1DIApACYAQuYukl9ZATwDKjdAGFUtAK4sGIM6SuADJ4DJgAcj4ARpjEEgBspAAOqAqETgwe3r7%2ByanpAiFhkSwxcVyJdpgOGUIETMQEWT5%2BAVU1AnUNBEUR0bEJtvWNzTltwz2hfaUDFQCUtqhexMjsHOYAzKHI3lgA1CYbbsgIxAKoh9gmGgCCm9u7mAdHyAoE%2BIKX13e3b0yOyD2AH0gaFaFMQXs3ugQChTudYY42ECUqECIc3NDYSczgxULCFABPN6YFhAnaoZAAa0uezxAHcIHMTAB2Ky3PZ7YiYAjLBhQ97Y%2BF42EIPDABBA7kKTxeRwCcm0SlU2EMpmHdl3FkAEW%2B3z%2BAOBoIY4LCkK8DDSwDC6D2SqMdoEwC56AICmQECxcNx%2BJASMwKNQaIxXpxCJARJJZIp1NpABUNHG8GxlEHBHMDmzvpymHLUHtwqh6c9tQKYd7w2KJVLMDLvPKGIrlaqi%2BqNprOdzecR%2BaHhb70Mt/hlyUw3iHBRWRSBmHilGgGOgFJcIIXiwBaPYJpMptMEOYAOjQFoIbc1rN1t2%2BaL2LCYoXVWY5nL2uaIewICEYJbpreZ7ezF8uz5F03Q9T9GH/c8dQ4BZaE4ABWXg/G4Xh8Q4NxrGsKElhWJ5Nh4UgCE0WCFhVDZ4gPMwAA5qI0GiNCkWizAATg0aR4I4SRkJI0h0N4BQQA0IiSIWOBYCQNAWCSOhYnISgpJk%2Bg4h2QxgCBAhiAtKk%2BDoAhYkEiAol4qJQgaQlOEI0zmGIQkAHkom0apiNQ0gpLYQQ7JNCyOC0UgsCiLxgDcMRaEE1ysDvIxxF83h8G5GoADda14zBVGqOU1kItFME4vzwSiYhzI8LBeM05NLN4ZLiCiVJMG1Uk1NNUBYoWKgDGABQADU8Ewek7KSRhKpkQQRDEdgpBG%2BQlDUXjdC4fQ1JQLDLH0PAokEyAFlQJIG3CtDquIPAsE2plbFy5yMhcRdRj8BbgimEoyj0FI0gbW6XvyBteie2ZzvsBsuhGWUcgW9pAYmH7%2BnKIZug%2BsHIce6GJAWGVllWFH9EQnjYr4zg9lUaj4nXeJJD2VSHQgTTtIzCBcEIEgDjMDYuDmXgXK0OYyJADYNgPFkKhY2j4i4DR4johCzA4zhuNIFC/P42whJE1rSHExAUFQaTZLICgIEUnWQAaFhEuo9cKeAddqYYHSaFofTiEM4zces8zhtd2yHKchxhvcxgCC82gfL8gKgpC2gwuGyKmrWEO8ASxxkvCvy0oy/ThpyvLeAKorbJK2P2aOlhhuq2qlAaqLgGa0S%2BA67rev6waUMI/hRtEcRJtb6aVHUXHdACAwjGWyxrDWjb4G23aMn2vjDuOlKtv%2By7nAgVx4cCRcoZmGHXoKTIQbuvI3oyLfnrBi6OgYIGmgPvRwdqRHimRhG4dvl/GlP2ZUdwjHWaxjgkJy14uhV8xATZmyYOTQezoqZaRtrTemRBiBMxZmzFWnNuasQPCxFikgha8xZJISQFEEIsn/rLeWaFOACWVhzUi/8zC8GLqLYSlC8YcHZqJBY1U0jOEkEAA%3D) and is more accurate -- so that the periodic serializer check happens at the specified intervals instead of something similar but unpredictable.

Whether this precision or system portability is necessary or only adds unneeded system overhead is debatable. If you decide it's not worthwhile it should be straightforward to switch back to the `TSCTimer` in `CaenBase.cpp`.

Reply:
Thanks for the description. I support your point on portability and actually the more predictable triggering also seems practical. I discuss this with @mortenjc but I prefer to switch over to this new method.

Reply:
I wanted a fast, imprecise timer that could be used in a tight loop with no overhead. Look at this 
LinkedIn article I wrote for some of the motivation behind.

https://www.linkedin.com/pulse/synchronising-time-across-xeon-cores-morten-jagd-christensen/?trackingId=Z2MJRMI8Sd2iM9l5q%2FMfEw%3D%3D

I am still undecided on whether to move to a new timer since we have no requirement  of portability to other architectures at the moment and no requirement in this case to have a precise time for periodic tasks.

Reply:
Looks like CheckTimer class could be replaced by Timer class as both are using  std::chrono high_resolution_clock
perhaps with small modifications. So we could let this one be for now and decide later.

Reply:
@mortenjc , your mean that we accept CheckTimer and we merge it either Timer or change back to TSCTimer? All fine for me, but just create a ticket in JIRA for the task for later.